### PR TITLE
Added pre_query and post_query for #76 and get() for #84

### PR DESCRIPTION
**Pull request by coder3112**
I have added the following things for #76 :
- ``` pre_insert```
- ``` post_insert```
- ``` pre_update```
- ``` post_update```
- ``` pre_delete```
- ``` pre_delete```
All tests passed locally for both sqlite and postgres.
Will make new tests soon.

### CODE REVIEW

**Comment by coder3112** [comment]
I still need to allow for arguments and async functions. These will be coming soon

**Comment by dantownsend** [comment]
@coder3112 Thanks for this. I'll try and review ASAP.

**Comment by coder3112** [comment]
Ok, so I have also added the fix for #77 in this PR.
Realize its a mistake but did not realize what I was doing. (Should have created a new branch as it is separate issue)

File: piccolo/query/methods/update.py [hunk 1]
```diff
@@ -32,15 +33,27 @@ def where(self, where: Combinable) -> Update:
...
         for column, value in self.values_delegate._values.items():
             if len(column._meta.call_chain) > 0:
-                raise ValueError(
-                    "Related values can't be updated via an update"
-                )
+                raise ValueError("Related values can't be updated via an update")
+
+    async def run_pre_functions(self):
+        for function in self.table._meta.pre_update:
+            function()
```
Comment (aminalaee) [inline piccolo/query/methods/update.py:44]: @coder3112 This is more of a question, shouldn't these be awaited?

Comment (coder3112) [reply piccolo/query/methods/update.py:44]: Good Catch! I just ran it and yes you are right. A couple changes need to be made so i guess I'll close this.

**Pull request closed without merging**